crates/blockchain-tree/src/block_indices.rs

```diff
@@ -21,13 +21,19 @@ pub struct BlockIndices {
     /// `number_to_block` as those are chain specific indices.
     canonical_chain: CanonicalChain,
     /// Index needed when discarding the chain, so we can remove connected chains from tree.
-    /// NOTE: It contains just a blocks that are forks as a key and not all blocks.
+    /// NOTE: It contains just blocks that are forks as a key and not all blocks.
     fork_to_child: HashMap<BlockHash, HashSet<BlockHash>>,
+    /// Utility index for Block number to block hash(s).
+    ///
+    /// This maps all blocks with same block number to their hash.
+    ///
+    /// Can be used for RPC fetch pending block(s) in chain by its number.
```

Comment:
This part can be incorrect, you can have a block of the same number but it does not mean it is extending the canonical chain maybe it is just a fork, pending blocks are considered only the blocks that extend it.

Reply:
> pending blocks are considered only the blocks that extend it.

got it, will make that clear